beacon-chain/blockchain/receive_attestation.go

```diff
@@ -111,6 +111,16 @@ func (s *Service) spawnProcessAttestationsRoutine() {
 	}()
 }
 
+// saturatingReorgInterval makes sure that the reorg interval cannot be a negative number
+// it should be saturated to 0 if it's the case
+func saturatingReorgInterval() time.Duration {
```

Comment:
Rather than setting this as zero, it might be better to set the following below as a dynamic value:
```
const reorgLateBlockCountAttestations = 2 * time.Second
```
Ex: If 2 seconds maps to 12 second slots, then for 1 second slots it should be 160ms. Otherwise the reorg interval starts breaking for non-standard configs.

cc @potuz might have some thoughts since this was introduced by him

Reply:
yeah, we had discussed that this constant would be irrelevant on non-mainnet networks because we wouldn't be reorging late blocks, but now this constant becomes a higher issue if we're removing the flag to disable this behavior. I am not sure what is the right approach here. Computing this to be `SecondsPerSlot / 2 / intervalsPerSlot` would produce the right value for mainnet. I suspect that any network that tries to have short blocks would be very broken by the reorg feature.